services/shotgrid_common/ayon_shotgrid_hub/__init__.py

```diff
@@ -306,6 +306,16 @@ def react_to_shotgrid_event(self, sg_event_meta):
                 f"Ignoring event, AYON project {self.project_name} not found.")
             return
 
+        # task revived only sends attribute_change event, before attribute
+        # change of parent asset AND finally entity_revival of Asset event
+        if (sg_event_meta["type"] == "attribute_change"
+            and sg_event_meta["attribute_name"] == "retirement_date"):
+            if sg_event_meta["entity_type"].lower() != "task":
```

Comment:
The `retirement_date` change event I assume is ALSO triggered if you were to retire a task instead of reviving it.

Should we check the value of the retirement date being set?

Looking at [this](https://community.shotgridsoftware.com/t/retirement-dates/10545) my assumption is that the retirement date gets cleared out maybe when reviving? Should we hence also check the value whether it's really retiring it or not? Or do we want to handle the retiring in the same manner?

Or, do we even _care_ about processing the retirement date attribute events? Or are we actually syncing that shotgrid data too?

Reply:
Updated logic a bit.